### PR TITLE
feat(local_visualizer.py): add check dir exist

### PR DESCRIPTION
**Pull request by tpoisonooo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

add output dir check, `output_file` is actually an directory.

https://github.com/open-mmlab/mmdeploy/pull/1039 needs it.

## Modification

add `os.mkdir`

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/visualization/local_visualizer.py [hunk 1]
```diff
@@ -605,6 +605,9 @@ def add_datasample(self,
                 win_name=name,
                 wait_time=wait_time)
 
+        if not osp.exists(out_file):
```
Comment (ZwwWayne) [inline mmdet3d/visualization/local_visualizer.py:608]: can we use mmengine.mkdir_or_exist?

Comment (tpoisonooo) [reply mmdet3d/visualization/local_visualizer.py]: done.

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
Lint failed, can be merged after fix linting issue.

**Comment by tpoisonooo** [comment]
> Lint failed, can be merged after fix linting issue.

done.

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**